pkg/image/admission/imagestream_limits_test.go

```diff
@@ -0,0 +1,420 @@
+package admission
+
+import (
+	"strings"
+	"testing"
+
+	kapi "k8s.io/kubernetes/pkg/api"
+	"k8s.io/kubernetes/pkg/api/resource"
+	"k8s.io/kubernetes/pkg/client/cache"
+	fake "k8s.io/kubernetes/pkg/client/unversioned/testclient"
+
+	imagetest "github.com/openshift/origin/pkg/image/admission/testutil"
+	imageapi "github.com/openshift/origin/pkg/image/api"
+	quotautil "github.com/openshift/origin/pkg/quota/util"
+)
+
+func TestGetMaxLimits(t *testing.T) {
+	for _, tc := range []struct {
+		name           string
+		lrs            []kapi.LimitRange
+		expectedLimits kapi.ResourceList
+	}{
+		{
+			name: "no limit range",
+		},
+
+		{
+			name: "unrelevant limit range",
+			lrs: []kapi.LimitRange{
+				{
+					Spec: kapi.LimitRangeSpec{
+						Limits: []kapi.LimitRangeItem{
+							{
+								Type: kapi.LimitTypePod,
+								Max:  kapi.ResourceList{kapi.ResourceCPU: resource.MustParse("200m")},
+							},
+						},
+					},
+				},
+			},
+		},
```

Comment:
You could create helper methods that will allow you to inject just the Limits to make the tests easier to read.